src/ResultUtils.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace PowerUtils.Results
+{
+#if NET6_0_OR_GREATER
+    public partial record struct Result : IResult
+#else
+    public partial record Result : IResult
+#endif
+    {
+        /// <summary>
+        /// Creates a success result
+        /// </summary>
+        public static Result Ok() => new();
+
+        /// <summary>
+        /// Creates a success result
+        /// </summary>
+        public static Result<TValue> Ok<TValue>(TValue value) => Result<TValue>.Ok(value);
+
+        /// <summary>
+        /// Creates an <see cref="Result"/> from an <see cref="IError"/>
+        /// </summary>
+        public static Result From(IError error) => new(error);
+
+        /// <summary>
+        /// Creates an <see cref="Result{TValue}"/> from an <see cref="IError"/>
+        /// </summary>
+        public static Result<TValue> From<TValue>(IError error) => Result<TValue>.From(error);
+
+        /// <summary>
+        /// Creates an <see cref="Result{TValue}"/> if <see cref="IEnumerable{IError}"/> is empty or null. If not create an error result
+        /// </summary>
+        public static Result<TValue> Create<TError, TValue>(IEnumerable<TError> errors, Func<TValue> onCreator)
+            where TError : IError
+        {
+            if(errors is null || !errors.Any())
```

Comment:
if the errors list is null, won't `.Any()` trow and exception?

Reply:
No, because before you have a check for null `if(errors is null...`

Reply:
yes, that will catch the problem, that i was pointing..
Initially tought that it could cause a NullArgumentException, because the tests only used already initialized lists, but nevermind